src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -705,19 +705,20 @@ def create_service_principal(cmd, identifier):
 
 def _create_service_principal(cli_ctx, identifier, resolve_app=True):
     client = _graph_client_factory(cli_ctx)
-
+    app_id = identifier
     if resolve_app:
         if _is_guid(identifier):
             result = list(client.applications.list(filter="appId eq '{}'".format(identifier)))
         else:
             result = list(client.applications.list(
                 filter="identifierUris/any(s:s eq '{}')".format(identifier)))
 
-        if not result:  # assume we get an object id
-            result = [client.applications.get(identifier)]
-        app_id = result[0].app_id
-    else:
-        app_id = identifier
+        try:
+            if not result:  # assume we get an object id
+                result = [client.applications.get(identifier)]
+            app_id = result[0].app_id
+        except GraphErrorException:
+            pass  # fallback to appid (maybe from an external tenant?)
```

Comment:
How does this work? I would have assumed that the SP would have to somehow more directly reference the App, if it's not in the sub? Does AD just "find" the App ID at runtime, even though the CLI doesn't see it?

Reply:
That is correct. AAD would be able to find it as long as you have configured that APP as `multi-tenant` in the original tenant.
CLI used to error out when can't find it in the current tenant, but now let service end do the last try. If the APP-ID is not right, users still get an error.

Reply:
Cool. Thanks for the details!